### PR TITLE
Add support for openstack provider (#1103)

### LINKED ISSUE #1103: Terraform scan - OpenStack provider ?
Hello,

Do you plan to support Terraform scans with OpenStack in the future ?

Regards

### PR DESCRIPTION
**Pull request by pascal-hofmann**
This PR adds support for openstack and closes #1103.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by pascal-hofmann** [comment]
There were some bogus / missing tests. These are fixed / added now. Ready for review.

**Review by schosterbarak — APPROVED** [review]

**Review by gruebel — COMMENTED** [review]
thanks for the great addition 🚀, I just found some minor things

File: checkov/terraform/checks/provider/openstack/credentials.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        id = "CKV_OPENSTACK_1"
+        supported_provider = ["openstack"]
+        categories = [CheckCategories.SECRETS]
+        super().__init__(name=name, id=id, categories=categories, supported_provider=supported_provider)
+
+    def scan_provider_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
+        """
+        see: https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs#configuration-reference
+        """
+        if self.secret_found(conf, "password"):
```
[the remark this answers is not in the document]
Comment (nimrodkor) [reply checkov/terraform/checks/provider/openstack/credentials.py:19]: To protect from `None` values, I'd suggest something like this:
```suggestion
        if conf.get("password"):
```

[the remark this answers is not in the document]
Comment (gruebel) [reply checkov/terraform/checks/provider/openstack/credentials.py:19]: good point, I like that better

[the remark this answers is not in the document]
Comment (pascal-hofmann) [reply checkov/terraform/checks/provider/openstack/credentials.py]: I amended my commit with the suggested changes.

File: checkov/terraform/checks/resource/openstack/AbsSecurityGroupUnrestrictedIngress.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.common.util.type_forcers import force_list
+from checkov.common.util.type_forcers import force_int
+
+
+class AbsSecurityGroupUnrestrictedIngress(BaseResourceCheck):
+    def __init__(self, check_id, port):
+        name = "Ensure no security groups allow ingress from 0.0.0.0:0 to port %d (tcp / udp)" % port
```
[the remark this answers is not in the document]
Comment (pascal-hofmann) [reply checkov/terraform/checks/resource/openstack/AbsSecurityGroupUnrestrictedIngress.py]: I amended my commit with the suggested changes.

**Review by nimrodkor — APPROVED** [review]

**Comment by nimrodkor** [comment]
thanks @pascal-hofmann !!

**Comment by pascal-hofmann** [comment]
Thanks for the quick merge!

**Comment by MindTooth** [comment]
It's great to see OpenStack support, but saying that no hardcoded value is somewhat lacking and then not provide the best approach.

My use case is that I use a variable that's exported via an environment variable or a `terraform.tfvars`.  Meaning, no hardcoded value exists.  `application_credential_secret = var.os_application_credential_secret` is not hardcoded, IMHO.  If this is not enough, I'm eager to learn why.

---

I want to be absolutely clear, I love that OpenStack is supported.  I work with OS almost daily.  Thank you. 👍🏻

**Comment by gruebel** [comment]
@MindTooth great question, you can expose them through plain old environment variables and don't need to reference them in the provider

- `token` -> `OS_TOKEN` or `OS_AUTH_TOKEN`
- `password` -> `OS_PASSWORD`
-  `application_credential_secret` -> `OS_APPLICATION_CREDENTIAL_SECRET` fun fact this is not documented in the Terraform provider, but I took a look into source code 😄 

I hope that helps 🙂

**Comment by MindTooth** [comment]
Just wanted to drop by and let you know that the following works great:

- `OS_AUTH_TYPE=v3applicationcredential`
- `OS_AUTH_URL=`
- `OS_APPLICATION_CREDENTIAL_ID=`
- `OS_APPLICATION_CREDENTIAL_SECRET=`

I've converted almost all my projects to use only these.  **Note!**  Important that no other `OS_` variables are set.  As application credentials are scoped, they only work for one project.  Also, this removes the need for a `provider "openstack" {}` block.

**Pull request merged**